tests/unit/browser/urlmarks.py

```diff
@@ -0,0 +1,127 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2018 Ryan Roden-Corrent (rcorre) <ryan@rcorre.net>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for the global page history."""
+
+from unittest import mock
+import pytest
+from PyQt5.QtCore import QUrl
+
+from qutebrowser.browser import urlmarks
+
+
+@pytest.fixture
+def bm_file(config_tmpdir):
+    bm_dir = config_tmpdir / 'bookmarks'
+    bm_dir.mkdir()
+    bm_file = bm_dir / 'urls'
+    return bm_file
+
+
+def test_init(bm_file, fake_save_manager):
+    bm_file.write('\n'.join([
+        'http://example.com Example Site',
+        'http://example.com/foo Foo',
+        'http://example.com/bar Bar',
+        'http://example.com/notitle',
+    ]))
+
+    bm = urlmarks.BookmarkManager()
+    fake_save_manager.add_saveable.assert_called_once_with(
+        'bookmark-manager',
+        bm.save,
+        mock.ANY,  # TODO: compare signal argument for equality
```

Comment:
Passing `bm.changed` doesn't work?

Reply:
```
At index 2 diff: <bound PYQT_SIGNAL changed of BookmarkManager object at 0x7fad65b284c8> != <bound PYQT_SIGNAL changed of BookmarkManager object at 0x7fad65b284c8>
```

Forgot I had that `TODO` there though, I can remove it if we don't know how to compare them ...

Reply:
Huh, weird! It looks like you got it to work now though?

Reply:
Ah crap, I didn't mean to include this in the last commit, I thought I patched it out! Did the tests pass? They fail for me

Reply:
Weird! Yeah, they seem to pass just fine on Travis (though everything else is falling apart, like usual 😆). Can you try running tox with `-r` once to make sure you're on the newest versions of everything?